packages/notify-client/src/controllers/engine.ts

```
@@ -1213,4 +1219,34 @@ export class NotifyEngine extends INotifyEngine {
       })
     );
   };
+
+  private watchLastWatchedAccountIfExists = async () => {
+    // If an account was previously watched
+    if (this.client.lastWatchedAccount.keys.length === 1) {
+      const { lastWatched: account } =
+        this.client.lastWatchedAccount.get("lastWatched");
+
+      try {
+        // Account for invalid state where the last watched account does not have an identity.
+        const identity = await this.client.identityKeys.getIdentity({
+          account,
+        });
+        if (!identity)
+          throw new Error("No identity key for lastWatchedAccount");
+      } catch (error) {
+        this.client.logger.error(
+          `[Notify] Last watched account ${account} has no registered identity. Can not watch subscriptions by default. `
```

This is assuming the caught error is the explicitly defined case, and swallowing the actual `error` param.

```suggestion
          `[Notify] Engine > watchLastWatchedAccountIfExists failed: ${error}`
```